scipy/stats/_continuous_distns.py

```diff
@@ -5198,8 +5199,280 @@ def _entropy(self, beta):
+              are equivalent to ``f0``, and ``fb`` and ``fix_b`` are
+              equivalent to ``f1``.
+
+            - optimizer : The optimizer to use.  The optimizer must take ``func``,
+              and starting position as the first two arguments,
+              plus ``args`` (for extra arguments to pass to the
+              function to be optimized) and ``disp=0`` to suppress
+              output as keyword arguments.
+
+        Returns
+        -------
+        shape : tuple of floats
+            MLEs for any shape statistics.
+
+        Notes
+        -----
+        This fit is computed by maximizing a log-likelihood function, with
+        penalty applied for samples outside of range of the distribution. The
+        returned answer is not guaranteed to be the globally optimal MLE, it
+        may only be locally optimal, or the optimization may fail altogether.
+        """
+        kwds['floc'] = 0
+        kwds['fscale'] = 1
+        return super(rv_arbitrary, self).fit(data, *args, **kwds)[:-2]
+
+    def fit_loc_scale(self, data, *args):
+        """
+        Not applicable for ``rv_arbitrary``
+        """
+        # rv_arbitrary uses fixed loc and scale in order to keep the pdf
+        # normalised.
+        return 0, 1
+
+    def _unpack_loc_scale(self, theta):
+        # loc and scale fixed at 0, 1 for rv_arbitrary
+        return 0, 1, theta[0:self.numargs]
+
+    def _construct_default_doc(self, *args, **kwds):
+        pass
+
+    def _construct_doc(self, *args, **kwds):
+        pass
+
+
+def _get_shapes(funcs):
+    # using introspection get the name, number of positional arguments of a PDF
+    # function.
+    shapes_list = []
+    for func in funcs:
+        if func is None:
+            continue
+        shapes_args = _getargspec(func)  # NB: does not contain self
+
+        if len(shapes_args.args) < 1:
+            # have to supply at least 'x'
+            raise ValueError("The %f function you supplied must accept at least"
+                             "the 'x' argument" % func)
+        args = shapes_args.args[1:]  # peel off 'x'
+
+        if args:
+            shapes_list.append(args)
+
+            # *args or **kwargs are not allowed w/automatic shapes
+            if shapes_args.varargs is not None:
+                raise TypeError(
+                    '*args are not allowed w/out explicit shapes')
+            if shapes_args.keywords is not None:
+                raise TypeError(
+                    '**kwds are not allowed w/out explicit shapes')
+            if shapes_args.defaults is not None:
+                raise TypeError('defaults are not allowed for shapes')
+
+    if shapes_list:
+        shapes = shapes_list[0]
+
+        # make sure the signatures are consistent
+        for item in shapes_list:
+            if item != shapes:
+                raise TypeError('Shape arguments are inconsistent.')
+    else:
+        shapes = []
+
+    return shapes
```

Comment:
Same here: this is essentially a copy-paste of the logic from `rv_continuous`? If it's really needed here, this helper should be used in both places. Preferably though, none of this should be necessary outside of `rv_continuous.__init__`